File: src/main/java/org/chousik/collection/builder/collectors/ColorCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.Color;
import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

public class ColorCollector extends EnumCollector<Color> {
    @Override
    public Color ask(String name, IValidator<String> validator) throws ScriptExecutionError {
        return askEnum(name,validator, Color::valueOf, Color::getValue);
    }
}

File: src/main/kotlin/collection/builder/collectors/ColorCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.Color
import org.chousik.collection.validators.IValidator
class ColorCollector : EnumCollector<Color>() {

    override fun ask(name: String, validator: IValidator<String?>): Color {
        return askEnum(
            name, validator,
            { value: String -> Color.valueOf(value) }, Color.value
        )
    }
}